docs/philosophy.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Philosophy
+---
+
+Jekyll offers a unique philosophy when approaching the problem of static
+site generation. This core philosophy drives development and product
+decisions. When a contributor, maintainer, or user asks herself what Jekyll
+is about, the following principles should come to mind:
+
+### 1. No Magic
+
+Jekyll is not magic. A user should be able to understand the underlying
+processes that make up the Jekyll build without much reading. It should
+behave "as you'd expect."
```

Comment:
@parkr you've described this previously as "Jekyll does exactly what the user asks, no more, no less". Is that still accurate? "I expect this" is very different from "I asked you to do this" or "I understand what's going on under the hood to cause this to happen".